src/blueapi/service/model.py

```diff
@@ -157,6 +157,22 @@ class EnvironmentResponse(BlueapiBaseModel):
     )
 
 
+class RepositoryStatus(BlueapiBaseModel):
+    repository_name: str = Field(description="Name of the repository")
+    version: str = Field(description="Branch or tag of the repository")
+    is_dirty: bool = Field(description="Does the repository have uncommitted changes")
+
+
+class ScratchResponse(BlueapiBaseModel):
+    """
+    State of the scratch area.
+    """
+
+    package_info: list[RepositoryStatus] = Field(
+        description="package information", default=[]
```

Comment:
```suggestion
    package_info: list[RepositoryStatus] = Field(
        description="package information", default_factory=list
```

Else the same empty list gets passed as a default to all instances